qiskit/circuit/quantumcircuit.py

```diff
@@ -1261,7 +1262,11 @@ def _append(
 
         return instruction
 
-    def _update_parameter_table(self, instruction: Instruction) -> Instruction:
+    def _update_parameter_table(self, instruction: Operation) -> Operation:
+        # A generic Operation object at the moment does not require to have params.
```

Comment:
I have updated the code to check `isinstance(instruction, Instruction)`. 

I don't remember the reason of not including `params` in `Operation` either. Possibly this would be a good idea, even though `Cliffords` don't use `params` to store stabilizer/destabilizer table. Do we want them to? Is it true that two `Instructions` with the same `name`, `num_qubits`, `num_clbits` and parameters are completely equivalent? Do we want this to be true?